ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -586,4 +662,74 @@ void resetStore() {
       operationReceived = null;
     }
   }
+
+  /**
+   * An extension of {@link ReplicationManager} to help with testing.
+   */
+  private static class MockReplicationManager extends ReplicationManager {
+
+    // Variables for controlling and examining the values provided to controlReplicationForPartitions()
+    Boolean controlReplicationReturnVal;
+    List<? extends PartitionId> idsVal;
+    List<String> originsVal;
+    Boolean enableVal;
+
+    /**
+     * Static construction helper
+     * @param verifiableProperties the {@link VerifiableProperties} to use for config.
+     * @param storageManager the {@link StorageManager} to use.
+     * @param clusterMap the {@link ClusterMap} to use.
+     * @param dataNodeId the {@link DataNodeId} to use.
+     * @return an instance of {@link MockReplicationManager}
+     * @throws ReplicationException
+     */
+    static MockReplicationManager getReplicationManager(VerifiableProperties verifiableProperties,
+        StorageManager storageManager, ClusterMap clusterMap, DataNodeId dataNodeId) throws ReplicationException {
+      ReplicationConfig replicationConfig = new ReplicationConfig(verifiableProperties);
+      ClusterMapConfig clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+      StoreConfig storeConfig = new StoreConfig(verifiableProperties);
+      return new MockReplicationManager(replicationConfig, clusterMapConfig, storeConfig, storageManager, clusterMap,
+          dataNodeId);
+    }
+
+    /**
+     * Constructor for MockReplicationManager.
+     * @param replicationConfig the config for replication.
+     * @param clusterMapConfig the config for clustermap.
+     * @param storeConfig the config for the store.
+     * @param storageManager the {@link StorageManager} to use.
+     * @param clusterMap the {@link ClusterMap} to use.
+     * @param dataNodeId the {@link DataNodeId} to use.
+     * @throws ReplicationException
+     */
+    MockReplicationManager(ReplicationConfig replicationConfig, ClusterMapConfig clusterMapConfig,
+        StoreConfig storeConfig, StorageManager storageManager, ClusterMap clusterMap, DataNodeId dataNodeId)
+        throws ReplicationException {
+      super(replicationConfig, clusterMapConfig, storeConfig, storageManager, stream -> null, clusterMap, null,
+          dataNodeId, null, clusterMap.getMetricRegistry(), null);
+      reset();
+    }
+
+    @Override
+    public boolean controlReplicationForPartitions(List<? extends PartitionId> ids, List<String> origins,
+        boolean enable) {
+      if (controlReplicationReturnVal == null) {
+        throw new IllegalStateException("Return val not set. Don't know what to return");
+      }
+      idsVal = ids;
+      originsVal = origins;
+      enableVal = enable;
+      return controlReplicationReturnVal;
+    }
+
+    /**
+     * Resets all state
+     */
+    void reset() {
```

Comment:
I like to leave stuff non-private in private classes to indicate that the parent class is using the functions. Having this as private or not has no real effect